app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -248,17 +249,28 @@ class WallpapersUseCases(
             // dispatched above, we may still need to update other metadata about it.
             store.dispatch(AppAction.WallpaperAction.UpdateCurrentWallpaper(currentWallpaper))
 
-            fileManager.clean(
-                currentWallpaper,
-                possibleWallpapers
-            )
-
             val wallpapersWithUpdatedThumbnailState = possibleWallpapers.map { wallpaper ->
                 val result = downloader.downloadThumbnail(wallpaper)
                 wallpaper.copy(thumbnailFileState = result)
             }
 
-            val defaultIncluded = listOf(Wallpaper.Default) + wallpapersWithUpdatedThumbnailState
+            val migratedWallpapers =
+                if (settings.shouldMigrateLegacyWallpapers) {
+                    settings.shouldMigrateLegacyWallpapers = false
+                    fileManager.migrateAndCleanLegacyWallpapers(
+                        remoteAvailableWallpapers = wallpaperList,
+                        localAvailableWallpapers = possibleWallpapers,
+                        downloadThumbnail = downloader::downloadThumbnail
+                    )
+                } else {
+                    fileManager.getMigratedWallpapers(
+                        remoteAvailableWallpapers = wallpaperList,
+                        localAvailableWallpapers = possibleWallpapers
+                    )
+                }
+
+            val defaultIncluded = listOf(Wallpaper.Default) + wallpapersWithUpdatedThumbnailState +
```

Comment:
Including `migratedWallpapers` here shouldn't be necessary, as there is not currently a requirement to display an expired wallpaper in the available wallpaper list in the settings page.

Reply:
Just to be sure this is the intended behavior: When testing this PR I noticed the migrated wallpaper (while still set) is not displayed in the wallpaper settings. Therefore if the user selects any other wallpaper the migrated one will be cleaned on the next startup. Here is a video of the behavior I described: 

https://user-images.githubusercontent.com/35462038/188904518-938e5480-7d96-407d-a223-7b63335cdb3b.mp4

Reply:
Yes, that behavior is intended (if suboptimal). During wallpapers V1 it was decided that users could keep their selected wallpaper once a promotion ended, but it would no longer appear in settings and if the wallpaper was switched the user would lose access to it. This was mostly a "this is the easiest thing to do for now" decision.

There was some talk about improving that in the future, such that once a user selected a wallpaper it would be available in their "collection" indefinitely. However, such an improvement I think is outside the scope of this issue and could instead be done during further iterations of the feature.

Totally unrelated: I spotted your deftones search - nice taste lol